src/execution_tree/compiler/compiler.cpp

```diff
@@ -466,11 +466,15 @@ namespace phylanx { namespace execution_tree { namespace compiler
                 snippets_.sequence_numbers_[name]++;
 
             // get global name of the component created
-            std::string global_name =
-                name + "$" + std::to_string(sequence_number);
+            primitive_name_parts name_parts;
+            name_parts.primitive = name;
+            name_parts.sequence_number = sequence_number;
+
             if (id.id >= 0)
             {
-                global_name += annotation(id);
+                name_parts.compile_id = snippets_.compile_id_ - 1;
```

Comment:
Same here, `compile_id` can always be set.

Reply:
Done.